src/smoltcp_impl/mod.rs

```diff
@@ -130,8 +130,16 @@ impl<'a> SocketSetWrapper<'a> {
         let mut sockets = self.0.lock();
         for item in sockets.iter_mut() {
             match item.1 {
+                Socket::Tcp(s) => {
+                    let local_addr = s.get_listen_endpoint();
```

Comment:
The `SO_REUSEADDR` should check the bound address rather than the listen endpoint.(see https://manpages.debian.org/unstable/manpages/socket.7.en.html#SO_REUSEADDR)

Reply:
done